Cannot find symbols including the MNIST dataset on Linux

The command `go test -v` in the container complains that it cannot find symbols including the MNIST dataset.  It is weird that `go test -v` works with macOS.

```
root@a483ce0b3e5d:/go/src/github.com/wangkuiyi/gotorch# go test -v
# github.com/wangkuiyi/gotorch
./cgotorch/libcgotorch.so: undefined reference to `torch::data::datasets::MNIST::MNIST(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, torch::data::datasets::MNIST::Mode)'
./cgotorch/libcgotorch.so: undefined reference to `c10::Symbol::fromQualString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
./cgotorch/libcgotorch.so: undefined reference to `c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
FAIL	github.com/wangkuiyi/gotorch [build failed]
```

Anyway, we can merge this PR and fix the problem in future PRs.

_Originally posted by @wangkuiyi in https://github.com/wangkuiyi/gotorch/pull/58#issuecomment-668981116_

--- a/cgotorch/Makefile.linux
+++ b/cgotorch/Makefile.linux
@@ -20,7 +20,8 @@ libcgotorch.so : cgotorch.cc cgotorch.h linux/libtorch
 	-o $@ -install_name @rpath/$@ \
 	-Wl,-rpath,libtorch/lib \
 	-Wl,-force_load libtorch/lib/libc10.so \
-	-lc10 -ltorch -ltorch_cpu
+	-lc10 -ltorch -ltorch_cpu \
+	-D_GLIBCXX_USE_CXX11_ABI=0
 
 clean:
 	rm -rf *.so

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -9,3 +9,4 @@ RUN cp /go/bin/* /usr/local/bin/
 
 RUN echo 'export GOPATH=/go' >> /root/.bashrc
 RUN echo 'export PATH=/usr/local/go/bin:/go/bin:$PATH' >> /root/.bashrc
+RUN echo 'export LD_LIBRARY_PATH=$GOPATH/src/github.com/wangkuiyi/gotorch/cgotorch/libtorch/lib'

